catalog/spatial/geocoding/spatial-geocoding-api/src/main/java/org/codice/ddf/spatial/geocoding/FeatureExtractor.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.spatial.geocoding;
+
+import org.opengis.feature.simple.SimpleFeature;
+
+/**
+ * A {@code FeatureExtractor} provides methods for extracting {@link SimpleFeature} objects from
+ * various resources.
+ */
+public interface FeatureExtractor {
+
+  /**
+   * Extracts geographic features from a resource as {@link SimpleFeature} objects and passes each
+   * through the callback {@code extractionCallback}. The callback is called exactly once for each
+   * {@code SimpleFeature} object extracted from the resource.
+   *
+   * @param resource identifier for the resource containing geographic features. the implementation
+   *     decides how to resolve it (filesystem, url, etc).
+   * @param extractionCallback the callback that receives each extracted {@code SimpleFeature}
+   *     object, must not be null
+   * @throws IllegalArgumentException if {@code extractionCallback} is null
+   * @throws FeatureExtractionException if an error occurs while extracting features from the
+   *     resource
+   */
+  void pushFeaturesToExtractionCallback(String resource, ExtractionCallback extractionCallback)
```

Comment:
I vote for `extractFeatures`

Reply:
I'm not sure I agree.  This actually does push the features into the extraction callback.  `pushFeatures` or `extractFeatures` is an incomplete description of what the method is actually doing.  FWIW, this method name was taken from https://github.com/codice/ddf/blob/master/catalog/spatial/geocoding/spatial-geocoding-api/src/main/java/org/codice/ddf/spatial/geocoding/GeoEntryExtractor.java#L63

Reply:
I get that it's pushing features to the callback, but if the callback is in the method signature, it seems redundant to include "`ToExtractionCallback`" in the name of the method. I'd argue that if one were to implement a method in `String` to check for matches to a `Pattern`, it should be named `matches(Pattern)` rather than `matchesPattern(Pattern)`; this seems like the same sort of scenario.

Besides that, it states first and foremost in the Javadoc that it _extracts features_; correct me if I'm wrong, but it seems that that is the primary point of the method while the `extractionCallback` seems to just give the programmer the ability to tack on more actions as features are extracted.

Note that this is a :pencil2:, so I'll approve even if you choose to keep the name as is. This is a pretty subjective judgement call either way.